File: src/main/java/br/com/teste/java/testebackend/domain/Veiculo.java
package br.com.teste.java.testebackend.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;


@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
@Builder
public class Veiculo implements Serializable {

    private static final long serialVersionUID = -3656431259068389491L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="id_veiculo", unique = true)
    private Long id;
    private String name;
    private String modelo;

    @ManyToOne(fetch = FetchType.EAGER)
    private Linha linha;
}

File: teste-back-end/src/entity/Veiculo.ts
import {
  Entity, PrimaryGeneratedColumn, Column, ManyToOne,
} from 'typeorm';
import Linha from './Linha';

@Entity()
export default class Veiculo {
  constructor(name: string, modelo: string, linha: Linha) {
    this.name = name;
    this.modelo = modelo;
    this.linha = linha;
  }

  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  name: string;

  @Column()
  modelo: string;

  @ManyToOne(() => Linha, (linha) => linha.veiculos)
  linha: Linha;
}
